### ISSUE TITLE
Incorrect quote_item_id saved on order items during multiple address checkout

### ISSUE THREAD

**Issue by zaximus84** [magento/magento2#18349]
### Preconditions
1. Magento 2.2.6
2. Multiple address checkout enabled
3. quote_item.item_id and quote_address_item.address_item_id auto increments are different. (If they are the same, the values are correct on order placement due to sheer luck)

### Steps to reproduce
1. Add an item to cart, log in (or create an account), and proceed to multiple address checkout.
2. Complete all steps of checkout and place order

### Expected result
1. All resulting order_item records in the database have a quote_item_id value that matches the item_id value of the original quote_item record. In other words, the order_item should point at the quote_item.

### Actual result
1. All resulting order_item records in the database have a quote_item_id value that matches the address_item_id value of the original quote_address_item record. In other words, the order item points at the quote_address_item instead of the quote_item.

### Additional Information
Magento\Multishipping\Model\Checkout\Type\Multishipping::_prepareOrder calls \Magento\Quote\Model\Quote\Item\ToOrderItem::convert for each quote address item. The convert method relies on mapping from fieldset.xml to determine which values to copy from the quote item. This is the relevant id field:

`<fieldset id="quote_convert_item">
...
    <field name="id">
        <aspect name="to_order_item" targetField="quote_item_id" />
    </field>`

Because the field name is "id", the object copy service (utilized in the convert method) calls getId() on $item. This is fine during a regular checkout when $item is an instance of quote item, but it's a problem during multiple address checkout when $item is an instance of quote address item. In that case, it returns the primary key from the quote_address_item table instead of the quote_item table, and ultimately produces an order item where quote_item_id points at the quote address item id. That will mean one of the following:

1) The referenced quote_item record will be from an unrelated quote.
2) The referenced quote_item record will not exist.
3) The referenced quote_item record will be correct by sheer luck because the quote_item and quote_item_address auto increments happened to be the same during the checkout process. This is unlikely with anything other than a fresh Magento installation.

The solution would be to handle quote address item instances differently and ensure that the quote_address_item.quote_item_id is mapped to sales_order_item.quote_item_id.

**Comment by magento-engcom-team** [comment]
Hi @zaximus84. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@zaximus84 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96251`, `MAGETWO-96252` were created

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
@zaximus84, thank you for your report.     
    
Unfortunately, we are archiving this ticket now as it did not get much attention from both Magento Community and Core developers for an extended period. This is done in an effort to create a quality, community-driven backlog which will allow us to allocate the required attention more easily.

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.3-develop` branch. Thank you for collaboration.
